File: src/main/java/de/cloudypanda/main/adventcalendar/config/AdventCalendarSubmitItemConfig.java
package de.cloudypanda.main.adventcalendar.config;

import lombok.*;
import org.bukkit.Material;

@NoArgsConstructor
@Getter
@Setter
@RequiredArgsConstructor
public class AdventCalendarSubmitItemConfig {
    private @NonNull Material material;
    private @NonNull Integer amount;
    private Integer durability;
}

File: src/main/java/de/cloudypanda/main/adventcalendar/config/AdventCalendarSubmitItemConfig.kt
package de.cloudypanda.main.adventcalendar.config;

import org.bukkit.Material

data class AdventCalendarSubmitItemConfig(
    val material: Material,
    val amount: Int,
    val durability: Int?,
    val name: String?,
    val enchants: List<AdventCalendarSubmitItemEnchantConfig> = ArrayList()
) {
    constructor() : this(Material.AIR, 0, null, null)
}
